File: lovometer.cpp
#include <iostream>
#include <string>
#include <cmath>
#include <cctype>
#include "unistd.h"


using namespace std;
void ToLower (string & Astring){
    for (char & letter : Astring){
        letter = tolower(letter);
    }
}

void presentation (){
	cout << "voici le lovover 9000 !" << endl;
	system ("timeout 1 > nul");
	cout << "pour savoir a quel point vous etes lie avec une une personne tapez votre nom et le sien !" << endl;
	system("timeout 1 > nul");
	cout << "la machine calculera automatiquement le pourcentage de compatibilite que vous avez avec cette personne !" << endl;
}

unsigned calcLove (const string & lover1, const string & lover2){
	float lovoScore = 0.0;
	const float KMaxLovoScore = lover1.length () + lover2.length ();

	if (lover1.length () == lover2.length ()){
		lovoScore = 0.75 * (lover1.length() + lover2.length ());
	}
	else if (lover1.length () > lover2.length ()){
		lovoScore = 0.75 * lover1.length ();
	}
	else{
		lovoScore = 0.75 * lover2.length ();
	}

	for (unsigned i (0); i < lover1.length (); ++i){
		for (unsigned j (0); j < lover2.length (); ++j){
			if (lover2[j] == lover1[i]){
                if (j == i){
                    lovoScore += 1;
                }
                else {
                    lovoScore = lovoScore + (1/fabs (j - i));
                }
			}
		}
	}

	if (lovoScore > KMaxLovoScore){
		lovoScore = lovoScore - 0.25 * KMaxLovoScore;
	}

	return int (lovoScore * 100 / KMaxLovoScore);
}

int main (){
	presentation ();
	string name1, name2;
	cout << "entrer votre nom: ";
	getline (cin, name1);
	cout << "entrer le nom de votre amour: ";
	getline (cin, name2);
	ToLower(name1);
	ToLower(name2);
	cout << "votre compatibilite avec " << name2 << " est de: " << calcLove(name1, name2) << "% !" << endl;
	return 0;
}

File: python/lovometer.py
#made by Percevase

#!/usr/bin/env python
#-*- coding: utf-8 -*-

#import des fonctions sleep et fonction absolue
from time import sleep
from math import fabs

#fonction pour faire la presentation du systeme de l'algo
def presentation ():
	print ("voici le lovo9000 !")
	#sleep (2)
	print ("pour savoir a quel point vous etes lie a une personne entrer votre prenom et le sien !")
	print ("la machine calculera automatiquement le pourcentage de compatibilite que vous avez avec cette personne !")
	#sleep (2)
	return

#algorithme de calcule du pourcentage de compatibilite
def calcLove (lover1, lover2):
	
	lovoScore = 0
	KMaxLovoScore = len (lover1) + len (lover2)
	
	#application de coefficients pour booster un peu les pourcentages
	if len (lover1) == len (lover2):
		lovoScore = 0.75 * (len (lover1) + len (lover2))
	elif len (lover1) > len (lover2):
		lovoScore = 0.75 * len (lover1)
	else :
		lovoScore = 0.75 * len (lover2)
	
	for letterInTheFirstName in lover1:
		for letterInTheSecondName in lover2:
			if letterInTheSecondName == letterInTheFirstName:
				if lover2.index (letterInTheSecondName) == lover1.index (letterInTheFirstName):
					lovoScore += 1
				else :
					lovoScore = lovoScore + (1/fabs (lover2.index (letterInTheSecondName) - lover1.index (letterInTheFirstName)))
	
	#teste qui evite que le score depasse les 100%
	if lovoScore > KMaxLovoScore:
		lovoScore = lovoScore - 0.25 * (len (lover1) + len (lover2))
		
	return lovoScore*100/KMaxLovoScore
	
#fonction qui sert a faire tourner l'algorithme
def main ():
	presentation ()
	name1 = input ("entrer votre prenom: ")
	name1 = name1.lower ()
	name2 = input ("entrer le prenom de votre amour: ")
	name2 = name2.lower ()
	print ("votre compatibilite avec " + name2 + " est: " + str (int (calcLove (name1, name2))) + "% !")
	return 
	
main ()

#made by Percevase
